packages/teams-js/src/internal/communication.ts

```diff
@@ -605,6 +614,46 @@ function updateRelationships(messageSource: Window, messageOrigin: string): void
 
 const handleParentMessageLogger = communicationLogger.extend('handleParentMessage');
 
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function retrieveMessageUUIDFromResponse(response: MessageResponse): MessageUUID {
+  if (response.uuid) {
+    const responseUUID = new MessageUUID(response.uuid);
+    const callbackUUID = retrieveMessageUUIDFromCallback(CommunicationPrivate.callbacks, responseUUID);
+    if (callbackUUID) {
+      return callbackUUID;
+    }
+    const promiseCallbackUUID = retrieveMessageUUIDFromCallback(CommunicationPrivate.promiseCallbacks, responseUUID);
+    if (promiseCallbackUUID) {
+      return promiseCallbackUUID;
+    }
+    const portCallbackUUID = retrieveMessageUUIDFromCallback(CommunicationPrivate.portCallbacks, responseUUID);
+    if (portCallbackUUID) {
+      return portCallbackUUID;
+    }
+  }
+  return CommunicationPrivate.legacyMessageIdsToUuidMap[response.id];
+}
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function retrieveMessageUUIDFromCallback(
```

Comment:
A comment explaining why this function is needed would be helpful. It's not immediate obvious to people who aren't familiar with Javascript Maps and using objects as keys in them why something like this would be needed instead of just accessing the map the "normal" way.